sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamIOSourceRel.java

```diff
@@ -45,15 +46,20 @@ public BeamIOSourceRel(
   }
 
   @Override
-  public PTransform<PInput, PCollection<Row>> buildPTransform() {
+  public PTransform<PCollectionList<Row>, PCollection<Row>> buildPTransform() {
     return new Transform();
   }
 
-  private class Transform extends PTransform<PInput, PCollection<Row>> {
+  private class Transform extends PTransform<PCollectionList<Row>, PCollection<Row>> {
```

Comment:
This is arguably `PTransform<PBegin, PCollection<Row>>`. But I can see how that isn't really different then a `PCollectionList<Row>` with zero elements. I guess I just don't like the `getPipeline().begin()` pattern everywhere, but this does simplify things overall.